### PR TITLE
SCI-O Model

### PR DESCRIPTION
**Pull request by gnanakeethan**
### Types of changes
- [ ] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### SCI-O Model

Implementation of SCI-O Model according to IEF Specifications. 

Accepts "grid-ci" and "energy" and returns "operational-emissions".

### CODE REVIEW

**Comment by narekhovhannisyan** [comment]
@gnanakeethan since we are keeping each model in it's appropriate folder, we can do the same with boavitza. Also if possible please add usage examples in `examples` folder so I can reuse them in cli tool

File: src/lib/sci-o/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if (!Array.isArray(observations)) {
+      throw new Error('observations should be an array');
+    }
+    observations.map((observation: KeyValuePair) => {
+      if ('grid-ci' in observation && 'energy' in observation) {
+        const grid_ci = parseFloat(observation['grid-ci']);
+        const energy = parseFloat(observation['energy']);
+        observation['operational-emissions'] = grid_ci * energy;
+      } else {
+        throw new Error(
```
Comment (jawache) [inline src/lib/sci-o/index.ts:22]: Separate error for each

File: src/lib/sci-o/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  calculate(observations: object | object[] | undefined): Promise<object[]> {
+    if (!Array.isArray(observations)) {
+      throw new Error('observations should be an array');
+    }
+    observations.map((observation: KeyValuePair) => {
+      if ('grid-ci' in observation && 'energy' in observation) {
+        const grid_ci = parseFloat(observation['grid-ci']);
+        const energy = parseFloat(observation['energy']);
+        observation['operational-emissions'] = grid_ci * energy;
```
Comment (jawache) [inline src/lib/sci-o/index.ts:20]: Does operation-emissions exist in the units.yml?

**Comment by jawache** [comment]
Agreed, let's add several usage examples. Full impl configuration files so we can test this on the cli.

File: src/lib/sci-o/index.ts (hunk 2 above)
Comment (gnanakeethan) [reply src/lib/sci-o/index.ts:20]: It was mentioned in the issue.

Comment (jawache) [reply src/lib/sci-o/index.ts]: I mean as you develop models, the units.yml needs also be updated in parallel. Will add it to the checklist to be made clear.

**Comment by jawache** [comment]
@gnanakeethan can you link to the source issue in every PR so we can easily make sure the requirements have been met.
Also for the source PR there are now checkboxes at the top, please take a second to tick the checkboxes to make clear you've implemented that feature.

image

**Comment by gnanakeethan** [comment]
@narekhovhannisyan @jawache I have added the documentation into the implementations folder.

**Review by jawache — APPROVED** [review]

**Pull request merged**